File: Work/WK/MS/Management/Management/Management/Management.Web/Modules/ServerTypes/BetHistory/PalyInfoForm.ts
import { StringEditor, IntegerEditor, DecimalEditor, DateEditor, BooleanEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface PalyInfoForm {
    PalyCurrentNum: StringEditor;
    PalyNum: StringEditor;
    PlayTypeId: IntegerEditor;
    LotteryId: IntegerEditor;
    UserName: StringEditor;
    NoteNum: IntegerEditor;
    SingleMoney: DecimalEditor;
    NoteMoney: DecimalEditor;
    NoteTime: DateEditor;
    IsWin: BooleanEditor;
    WinMoney: DecimalEditor;
    IsFactionAward: IntegerEditor;
    PlayTypeRadioId: IntegerEditor;
    RebatePro: DecimalEditor;
    RebateProMoney: StringEditor;
    WinNum: IntegerEditor;
    UserId: IntegerEditor;
    NoticeId: IntegerEditor;
    LotteryTime: DateEditor;
    UserRebatePro: DecimalEditor;
    Multiple: IntegerEditor;
    OrderKey: StringEditor;
    CurrencyUnit: DecimalEditor;
    Ratio: IntegerEditor;
    SourceType: StringEditor;
    MemoJson: StringEditor;
    ClientIp: StringEditor;
    RoomId: StringEditor;
    ResultJson: StringEditor;
}

export class PalyInfoForm extends PrefixedContext {
    static formKey = 'BetHistory.PalyInfo';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!PalyInfoForm.init)  {
            PalyInfoForm.init = true;

            var w0 = StringEditor;
            var w1 = IntegerEditor;
            var w2 = DecimalEditor;
            var w3 = DateEditor;
            var w4 = BooleanEditor;

            initFormType(PalyInfoForm, [
                'PalyCurrentNum', w0,
                'PalyNum', w0,
                'PlayTypeId', w1,
                'LotteryId', w1,
                'UserName', w0,
                'NoteNum', w1,
                'SingleMoney', w2,
                'NoteMoney', w2,
                'NoteTime', w3,
                'IsWin', w4,
                'WinMoney', w2,
                'IsFactionAward', w1,
                'PlayTypeRadioId', w1,
                'RebatePro', w2,
                'RebateProMoney', w0,
                'WinNum', w1,
                'UserId', w1,
                'NoticeId', w1,
                'LotteryTime', w3,
                'UserRebatePro', w2,
                'Multiple', w1,
                'OrderKey', w0,
                'CurrencyUnit', w2,
                'Ratio', w1,
                'SourceType', w0,
                'MemoJson', w0,
                'ClientIp', w0,
                'RoomId', w0,
                'ResultJson', w0
            ]);
        }
    }
}

File: Work/WK/MS/Management/Management/Management/Management.Web/Modules/BetHistory/PalyInfo/PalyInfoForm.cs
using Serenity.ComponentModel;
using Serenity.Web;
using System;

namespace Management.BetHistory.Forms
{
    [FormScript("BetHistory.PalyInfo")]
    [BasedOnRow(typeof(PalyInfoRow), CheckNames = true)]
    public class PalyInfoForm
    {
        public string PalyCurrentNum { get; set; }
        public string PalyNum { get; set; }
        public int PlayTypeId { get; set; }
        public int LotteryId { get; set; }
        public string UserName { get; set; }
        public int NoteNum { get; set; }
        public decimal SingleMoney { get; set; }
        public decimal NoteMoney { get; set; }
        public DateTime NoteTime { get; set; }
        public bool IsWin { get; set; }
        public decimal WinMoney { get; set; }
        public int IsFactionAward { get; set; }
        public int PlayTypeRadioId { get; set; }
        public decimal RebatePro { get; set; }
        public string RebateProMoney { get; set; }
        public int WinNum { get; set; }
        public int UserId { get; set; }
        public int NoticeId { get; set; }
        public DateTime LotteryTime { get; set; }
        public decimal UserRebatePro { get; set; }
        public int Multiple { get; set; }
        public string OrderKey { get; set; }
        public decimal CurrencyUnit { get; set; }
        public int Ratio { get; set; }
        public string SourceType { get; set; }
        public string MemoJson { get; set; }
        public string ClientIp { get; set; }
        public string RoomId { get; set; }
        public string ResultJson { get; set; }
    }
}